docs/user-guide/docker-compose.md

```diff
@@ -0,0 +1,380 @@
+
+In a working directory create a file called `docker-compose.yaml` with this content:
+
+```yaml
+version: '3.5'
+
+services:
+  ambassador:
+    image: quay.io/datawire/ambassador:0.38.0
+    ports:
+    # expose port 80
+    - 80
+    volumes:
+    # mount a volume where we can inject configuration files
+    - ./config:/ambassador/ambassador-config
+```
+
+Note the mounted volume. When Ambassador bootstraps on container startup it checks the `/ambassador/ambassador-config` directory for configuration files. We will use this behavior to configure ambassador.
+
+### Test using Ambassador's Diagnostics
+
+Run your compose environment and curl the diagnostics endpoint to ensure the compose file is working as expected.
+
+```bash
+# start your containers in the background
+docker-compose up -d
+
+# curl for the response header from the diagnostics endpoint
+curl -I localhost:8080/ambassador/v0/diag/
+
+# the response code should be 200
+HTTP/1.1 200 OK
+server: envoy
+date: Fri, 17 Aug 2018 21:07:37 GMT
+content-type: text/html; charset=utf-8
+content-length: 6459
+x-envoy-upstream-service-time: 10
+```
+
+## 2. Make a change to the default configuration
+
+Let's turn off the diagnostics page to demonstrate how we will enable and configure Ambassador.
+
+Create a `config` folder (which must match the mounted volume in the `docker-compose.yaml` file) and add a file called `ambassador.yaml` to the directory.
+(Note: Configuration files can have any name or combined into the same yaml file)
+
+```bash
+mkdir config
+touch config/ambassador.yaml
+```
+
+Set the contents of the `config/ambassador.yaml` to this yaml configuration:
+
+```yaml
+---
+apiVersion: ambassador/v0
+kind: Module
+name: ambassador
+config:
+  diagnostics:
+    # Stop the diagnostics endpoint from being publicly available
+    enabled: false
+```
+
+Now restart ambassador and test the diagnostics endpoint to ensure our configuration is in use:
+
+```bash
+# restart the container to pick up new configuration settings
+docker-compose up -d -V ambassador
+
+# curl the same diagnostics endpoint as the previous step
+curl -I localhost:8080/ambassador/v0/diag/
+
+# the response code should be 404
+HTTP/1.1 404 Not Found
+date: Fri, 17 Aug 2018 21:18:25 GMT
+server: envoy
+content-length: 0
+```
+
+Feel free to re-enable the diagnostics endpoint.
+
+## 3. Add a route mapping
+
+Now that we have demonstrated that we can modify the configuration let's add a mapping to route to `http://demo.getambassador.io/qotm/` service.
+
+Create a new file `config/mapping-qotm.yaml` with these contents:
+
+```yaml
+---
+apiVersion: ambassador/v0
+kind:  Mapping
+name:  qotm_mapping
+prefix: /qotm/
+rewrite: /qotm/
+service: demo.getambassador.io
+```
+
+Once again, restart ambassador and test the new mapping:
+
+```bash
+# restart the container to pick up new configuration settings
+docker-compose up -d -V ambassador
+
+# curl the quote-of-the-moment service
+curl localhost:8080/qotm/quote
```

Comment:
`curl localhost:8080/qotm/quote` is not accessible. Change to either `curl localhost:8080/qotm/quote/1` for select quote or `curl localhost:8080/qotm/` for random quote.